src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+// import { TodosContext } from '../../App';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { useTodos } from '../../utils/TodosContext';
 
 export const TodoModal: React.FC = () => {
+  // const { selectedTodoId, todos, setSelectedTodoId } = useContext(TodosContext);
+
+  const { todos, selectedTodoId, setSelectedTodoId } = useTodos();
+
+  const [user, setUser] = useState<User | undefined>();
+
+  const todo = todos.find(item => item.id === selectedTodoId) as Todo;
```

Comment:
Avoid `as` and it's better to store the whole selected todo and pass it as a prop to the modal when you should render the modal